framework/src/packets/ip/v6/srh.rs

```diff
@@ -120,48 +120,21 @@ impl Default for SegmentRoutingHeader {
 
 impl Header for SegmentRoutingHeader {}
 
-/// Type alias for a segment in segment routing header
-pub type Segment = Ipv6Addr;
-
 /// Error for a bad segment list
 #[derive(Debug, Fail)]
 #[fail(display = "Segment list length must be greater than 0")]
 pub struct BadSegmentsError;
 
-/// Error for bad index within segment list
-#[derive(Debug, Fail)]
-#[fail(display = "Index {} out of bounds of segment list length {}", _0, _1)]
-pub struct SegmentsOutOfBoundsError(usize, usize);
-
-/// Trait with helper functions for segment routing manipulation
-pub trait SegmentRoutingExt {
```

Comment:
not sure we should remove this, but after convo, I'll leave this up to others: @scrollins @pauljamescleary @clinedome if they found it helpful enough. It may not be though.

Reply:
I think I'm generally good with removing the helper functions and the magic (such as `set_segments_left_and_dst`) in favor of pulling that kind of functionality closer to the code that actually uses it. While these are general purpose, it's not clear that they're going to be helpful for any generic srh application as-is. I'm good with keeping all of this pretty basic and expecting folks to layer on their own abstractions if they see fit.